### PR TITLE
web/satellite: create ReferralStats component

### PR DESCRIPTION
**Pull request by VinozzZ**
What: 
Screen Shot 2019-06-24 at 10 41 32 AM

Why:
Need to display user's credit usage in referral program
Please describe the tests:
 - Test 1:
 - Test 2:
 
Please describe the performance impact:

## Code Review Checklist (to be filled out by reviewer)
 - [ ] Does the PR describe what changes are being made?
 - [ ] Does the PR describe why the changes are being made?
 - [ ] Does the code follow [our style guide](https://github.com/storj/docs/blob/master/code/Style.md)?
 - [ ] Does the code follow [our testing guide](https://github.com/storj/docs/blob/master/code/Testing.md)?
 - [ ] Is the PR appropriately sized? (If it could be broken into smaller PRs it should be)
 - [ ] Does the new code have enough tests? (*every* PR should have tests or justification otherwise. Bug-fix PRs especially)
 - [ ] Does the new code have enough documentation that answers "how do I use it?" and "what does it do?"? (both source documentation and [higher level](https://github.com/storj/docs), diagrams?)
 - [ ] Does any documentation need updating?
 - [ ] Do the database access patterns make sense?

### CODE REVIEW

File: web/satellite/src/components/referral/ReferralStats.vue [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+        font-family: 'Roboto';
+        font-style: normal;
+        font-weight: 500;
+        font-size: 14px;
+        line-height: 18px;
+      }
+
+      &__description {
+        display: block;
+        font-family: Roboto;
```
Comment (NikolaiYurchenko) [inline web/satellite/src/components/referral/ReferralStats.vue]: Why Roboto? Is this is a requirement from @tikhbana or DJ?

File: web/satellite/src/store/modules/usage.ts [hunk 2]
```diff
@@ -121,3 +121,25 @@ export const bucketUsageModule = {
         }
     }
 };
+
+export const creditUsageModule = {
+    state: {
+        creditUsage: { referred: 0, usedCredits: 0, availableCredits: 0 } as CreditUsage
+    },
+    mutations: {
+        [CREDIT_USAGE_ACTIONS.FETCH]: function(state: any, creditUsage: CreditUsage) {
```
Comment (NikolaiYurchenko) [inline web/satellite/src/store/modules/usage.ts]: just a nip - let`s define return type in ts

File: web/satellite/src/components/referral/ReferralStats.vue (hunk 1 above)
Comment (VinozzZ) [reply web/satellite/src/components/referral/ReferralStats.vue]: I will double check with them on the font

Comment (tikhbana) [reply web/satellite/src/components/referral/ReferralStats.vue]: Is this the referral program inside of the satellite GUI? If so, it should be Inter! Sorry, Figma glitches with the font sometimes. I'll fix that.

File: web/satellite/src/components/referral/ReferralStats.vue [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    methods: {
+        fetch: async function() {
+            const creditUsageResponse = await this.$store.dispatch(CREDIT_USAGE_ACTIONS.FETCH);
+            if (!creditUsageResponse.isSuccess) {
+                this.$store.dispatch(NOTIFICATION_ACTIONS.ERROR, 'Unable to fetch credit usage: ' + creditUsageResponse.errorMessage);
+            }
+        }
+    },
+    mounted() {
+        this.fetch();
```
Comment (crawter) [inline web/satellite/src/components/referral/ReferralStats.vue]: this may not be compiled
maybe (this as any).fetch()

**Pull request merged**